### PR TITLE
 Add test basement

### PR DESCRIPTION
**Pull request by jja725**

### CODE REVIEW

File: tests/test_read_range.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def validate_read_range(
         raise AssertionError(error_message)
 
 
-def test_invalid_read_range(
+def manual_test_invalid_read_range(
```
Comment (LuQQiu) [inline tests/test_read_range.py:65]: looks like we can finally modify this test to be automatic

**Comment by jja725** [comment]
@ChunxuTang any concern regarding the test in Uber env? if the docker image is a issue then we can make the docker test optional

**Comment by ChunxuTang** [comment]
> @ChunxuTang any concern regarding the test in Uber env? if the docker image is a issue then we can make the docker test optional

We don't need to worry about running the test in Uber's env for now. That's not a blocker for our changes.

File: environment_conda.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - aiohttp
+  - pytest=7.4.4
+  - pytest-timeout
+  - pytest-aiohttp
+  - requests
+  - humanfriendly
+  - mmh3
+  - sortedcontainers
+  - pip:
+        - git+https://github.com/kragniz/python-etcd3
```
[the remark this answers is not in the document]
Comment (jja725) [reply environment_conda.yaml:14]: the package is not available in the micromamba. Micromamba is a tiny python environment but sometimes it's too tiny

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9"]
```
[the remark this answers is not in the document]
Comment (jja725) [reply .github/workflows/ci.yml]: use 3.8,3.9 for now, we can add more if necessary

File: tests/test_read_range.py (hunk 1 above)
Comment (jja725) [reply tests/test_read_range.py:65]: no it's not. we need further modification to automate it

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          tests/
+
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
```
Comment (ChunxuTang) [inline .github/workflows/ci.yml:51]: 3.8 or 3.9? Which one is better? Your call.

**Review by ChunxuTang — APPROVED** [review]
LGTM. Just a very minor comment.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (jja725) [reply .github/workflows/ci.yml:51]: Let's stick with higher number which check more things

**Pull request merged**